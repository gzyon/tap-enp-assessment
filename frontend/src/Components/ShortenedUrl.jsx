import { Alert, Box, Button, IconButton, Link, Snackbar } from "@mui/material";
import React, { useState, forwardRef } from "react";
import ContentCopyIcon from '@mui/icons-material/ContentCopy';
import '@fontsource/jetbrains-mono/300.css';
import '@fontsource/jetbrains-mono/400.css';
import '@fontsource/jetbrains-mono/500.css';
import '@fontsource/jetbrains-mono/700.css';

const ShortenedUrl = (props) => {
    console.log(props);
    
    function handleClick() {
        if (props.shortUrl != undefined) {
            navigator.clipboard.writeText(props.shortUrl);
            props.setCopied(true);
            setInterval(() => {
                props.setCopied(false);
            }, 5000);
        }
    }

    return (
        <Box sx={{ width: '50%', justifyContent: 'center', alignItems: 'center', display: 'flex', flexDirection: 'row', border: 1, p: 1, borderRadius: '16px' }} >
            <Snackbar open={props.copied} autoHideDuration={5000}>
                <Alert security="info" sx={{width: '100%'}}>
                    Link Copied!
                </Alert>
            </Snackbar>
            <Link sx={{fontSize: 25, fontFamily: "Jetbrains Mono", 	pr: 2}} href={props.url} underline="none">
                {props.shortUrl}
            </Link>
            <IconButton onClick={handleClick} sx={{
            "&:hover, &.Mui-focusVisible": { borderColor: "yellow" }
            }} >
                <ContentCopyIcon sx={{color: "white"}} />
            </IconButton>
        </Box>
    )
}

export default ShortenedUrl;