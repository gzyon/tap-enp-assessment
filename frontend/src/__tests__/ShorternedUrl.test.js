import renderer from 'react-test-renderer';
import ShortenedUrl from '../Components/ShortenedUrl';

it('Renders a shortened URL referencing to the original URL', () => {
    const tree = renderer
        .create(<ShortenedUrl shortUrl={"abc123"} url={"https://drive.google.com/file/d/1XoERT9CyIk7mTT8qXvI6fvoCgjcaM2IL/view"} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});