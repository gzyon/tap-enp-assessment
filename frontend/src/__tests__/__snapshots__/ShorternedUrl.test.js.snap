// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a shortened URL referencing to the original URL 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-a1o7e9{width:50%;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;border:1px solid;padding:8px;border-radius:16px;}",
      }
    }
    data-emotion="css a1o7e9"
  />,
  <div
    className="MuiBox-root css-a1o7e9"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1hz99ly-MuiLink-root{-webkit-text-decoration:none;text-decoration:none;font-size:25px;font-family:Jetbrains Mono;padding-right:16px;}",
        }
      }
      data-emotion="css 1hz99ly-MuiLink-root"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1aslrwf-MuiTypography-root-MuiLink-root{margin:0;color:#1976d2;-webkit-text-decoration:none;text-decoration:none;font-size:25px;font-family:Jetbrains Mono;padding-right:16px;}",
        }
      }
      data-emotion="css 1aslrwf-MuiTypography-root-MuiLink-root"
    />
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1aslrwf-MuiTypography-root-MuiLink-root"
      href="https://drive.google.com/file/d/1XoERT9CyIk7mTT8qXvI6fvoCgjcaM2IL/view"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      abc123
    </a>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-gt47ji-MuiIconButton-root{text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-gt47ji-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-gt47ji-MuiIconButton-root:hover{background-color:transparent;}}.css-gt47ji-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}.css-gt47ji-MuiIconButton-root:hover,.css-gt47ji-MuiIconButton-root.Mui-focusVisible{border-color:yellow;}",
        }
      }
      data-emotion="css gt47ji-MuiIconButton-root"
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-tsklmn-MuiButtonBase-root-MuiIconButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-tsklmn-MuiButtonBase-root-MuiIconButton-root::-moz-focus-inner{border-style:none;}.css-tsklmn-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-tsklmn-MuiButtonBase-root-MuiIconButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-tsklmn-MuiButtonBase-root-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-tsklmn-MuiButtonBase-root-MuiIconButton-root:hover{background-color:transparent;}}.css-tsklmn-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}.css-tsklmn-MuiButtonBase-root-MuiIconButton-root:hover,.css-tsklmn-MuiButtonBase-root-MuiIconButton-root.Mui-focusVisible{border-color:yellow;}",
        }
      }
      data-emotion="css tsklmn-MuiButtonBase-root-MuiIconButton-root"
    />
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-tsklmn-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1d159sf-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;color:white;}",
          }
        }
        data-emotion="css 1d159sf-MuiSvgIcon-root"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
        data-testid="ContentCopyIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
        />
      </svg>
    </button>
  </div>,
]
`;
